lib/puppet/provider/network_vlan/nxapi.rb

```diff
@@ -0,0 +1,57 @@
+$LOAD_PATH.unshift(File.join(File.dirname(__FILE__),"..","..",".."))
+require 'puppet/type/cisco_vlan'
+require 'puppet/provider/cisco_vlan/nxapi'
+
+Puppet::Type.type(:network_vlan).provide(:nxapi, :parent => Puppet::Type.type(:cisco_vlan).provider(:nxapi)) do
```

Comment:
The inheritance is unnecessary as you're not calling anything from the parent class.